common/flatpak-transaction.c

```diff
@@ -4062,6 +4092,272 @@ flatpak_transaction_normalize_ops (FlatpakTransaction *self)
     }
 }
 
+static gboolean
+installed_ref_ref_equal (gconstpointer a,
+                         gconstpointer b)
+{
+  FlatpakInstalledRef *iref = (FlatpakInstalledRef *)a;
+  const char *needle_ref = (const char *)b;
+  g_autofree char *full_ref = flatpak_ref_format_ref (FLATPAK_REF (iref));
```

Comment:
Not sure its worth fixing, but this constant fomating fo the iref to a full ref is rather lame. We could easily split up the full ref once and do a cheaper comparison with that.

Reply:
Added a commit to fix that